Download CK library from compute-artifactory and link to Pytorch

akashveramd opened the pull request:
These are the changes made for this PR-

1. Removed generating and building CK kernels when building pytorch.
2. Download CK library from compute-artifactory and link with pytorch.
3. Enabled USE_CK_FLASH_ATTENTION based on USE_FLASH_ATTENTION option.

pruthvistony commented on CMakeLists.txt:
```diff
@@ -865,6 +865,13 @@ cmake_dependent_option(
   OFF)
 
+cmake_dependent_option(
+  USE_CK_FLASH_ATTENTION
```
We have environment variable - USE_CK_FLASH_ATTENTION, now we are adding cmake variable. is it requried??

pruthvistony commented on benchmarks/transformer/sdpa.py:
```diff
@@ -172,6 +172,7 @@ def print_results(experiments: List[Experiment]):
 
 
 def main():
+    torch.backends.cuda.preferred_rocm_fa_library("ck")
```
Why is this required? how was this working before?
is this for testing purpose?

akashveramd replied:
It was added for testing purpose. I will remove this.

pruthvistony commented on caffe2/CMakeLists.txt:
```diff
@@ -929,6 +929,10 @@ if(USE_ROCM)
   if(USE_FLASH_ATTENTION)
     target_link_libraries(torch_hip PRIVATE __caffe2_aotriton)
   endif()
+# link CK library
```
follow NIT

pruthvistony commented on CMakeLists.txt:
```diff
@@ -888,6 +895,13 @@ if(USE_ROCM)
 
+# CK shared lib linkage
+if(USE_ROCM)
+	if(UNIX AND (USE_CK_FLASH_ATTENTION))
```
follow NIT

pruthvistony commented on cmake/External/ck.cmake:
```diff
@@ -0,0 +1,44 @@
+#
```
follow spacing similar to other pytorch files. Dont use tabs.

Use lintrunner to verify and fix lint/NIT errors.

akashveramd commented:
Since I mistakenly created release/2.6_ck branch as a protected branch. Hence I cannot push my new commits added to address review comments. Therefore, I have created a new branch and a new PR which addresses the review comments. Here is the link to the new PR-
https://github.com/ROCm/pytorch/pull/2016

jithunnair-amd commented:
Closing in favor of https://github.com/ROCm/pytorch/pull/2016
